modules/ROOT/pages/intermediate-provisioning-tutorial.adoc

```diff
@@ -0,0 +1,168 @@
+
+* Add a systedmd dropin for auto login from serial console.
+* Disable automatic updates
+* Add a script at `/usr/local/bin/public-ipv4.sh` to run on boot
+* Configure a systemd service to run the script on first boot.
+
+== Write the Script
+
+[source,bash]
+----
+#!/bin/bash
+echo "Detected Public IPv4: is $(curl https://ipv4.icanhazip.com)" > \
+         /run/console-login-helper-messages/issue.d/30_public-ipv4.issue
+----
+
+This script uses https://icanhazip.com/[icanhazip.com] to update the issuegen from https://github.com/coreos/console-login-helper-messages[console-login-helper-messages] to output the node's public IPv4 address on the serial console during bootup.
+This is useful in cloud environments when you might have different public and private addresses.
+
+We'll store this script into `/usr/local/bin/public-ipv4.sh` when we provision the machine. We'll encode it into an FCCT config here shortly.
+
+=== Write the Systemd Service
+
+We need to call the script from the previous section by using a systemd unit.
+Here is one that works for what we want, which is to execute on first boot and not again:
+
+[source,service]
+----
+[Unit]
+Before=console-login-helper-messages-issuegen.service
+After=network-online.target
+ConditionPathExists=!/var/lib/issuegen-public-ipv4
+[Service]
+Type=oneshot
+ExecStart=/usr/local/bin/public-ipv4.sh
+ExecStartPost=/usr/local/bin/touch /var/lib/issuegen-public-ipv4
+RemainAfterExit=yes
+[Install]
+WantedBy=console-login-helper-messages-issuegen.service
+----
+
+We'll call this unit `issuegen-public-ipv4.service` and we'll embed it into the FCCT config in the next section.
+
+=== Write FCCT and convert to Ignition
+
+The final FCCT for what we want to do is shown below.
+We'll store these contents in `fcct-intermediate.yaml`.
+
+[source,yml]
+
+----
+variant: fcos
+version: 1.0.0
+systemd:
+  units:
+    - name: serial-getty@ttyeS0.service
+      dropins:
+      - name: autologin-core.conf
+        contents: |
+          [Service]
+          # Override Execstart in main unit
+          ExecStart=
+          # Add new Execstart with `-` prefix to ignore failure
+          ExecStart=-/usr/sbin/agetty --autologin core --noclear %I $TERM
+          TTYVTDisallocate=no
+    - name: issuegen-public-ipv.service
+      enabled: true
+      contents: |
+        [Unit]
+        Before=console-login-helper-messages-issuegen.service
+        After=network-online.target
+        ConditionPathExists=!/var/lib/issuegen-public-ipv4
+        [Service]
+        Type=oneshot
+        ExecStart=/usr/local/bin/public-ipv4.sh
+        ExecStartPost=/usr/bin/touch /var/lib/issuegen-public-ipv4
+        RemailAfterExit=yes
+        [Install]
+        WantedBy=console-login-helper-messages-issuegen.service
+storage:
+  files:
+    - path: /etc/zincati/config.d/90-disable-auto-updates.toml
+      contents:
+        inline: |
+          [updates]
+          enabled = false
```

Comment:
This is not used in this tutorial, it can be dropped.

Reply:
Hum, I would keep it as we do direct console login and an unexpected reboot at the beginning might confuse people?
On the other hand, we might consider dropping the whole 'disabling updates' as if people download a fresh FCOS image the likelihood that they will get a reboot during this tutorial is quite low. WDYT?

Reply:
Reading further, we have a section about automatic updates so we need to keep that. But to make this section work, we need to make sure users download an *older release*.